src/mds/MDCache.cc

```diff
@@ -1418,6 +1428,34 @@ void MDCache::adjust_subtree_after_rename(CInode *diri, CDir *olddir, bool pop)
   show_subtrees();
 }
 
+// ===================================
+// Cache Autotuning
+
+
+/*
+ * If mds_cache_autotune is set to true, then adjust cache_memory_limit using the rss usage periodically as per the mds_cache_resize_interval
+ */
+void MDCache::resize()
+{
+    if (cache_autotune) {
+      auto now = clock::now();
+      auto cache_resize_interval = g_conf().get_val<std::chrono::seconds>("mds_cache_resize_interval"); 
+      if (cache_memory_limit > mds->get_mds_memory_target())
+        // RSS usage has been experimentally shown to be around 25 percent more than the cache_memory_limit. Hence when the mds_memory_target is changed and the cache_memory_limit becomes more than the memory_target, the cache_memory_limit is reset as per the mds_memory_target(which is an upper bound on the rss usage) 
+        cache_memory_limit = (mds->get_mds_memory_target()*100)/125;
```

Comment:
I'm not sure why we have these above 3 lines. We should only need to scale `cache_memory_limit` to the memory target at startup. During resize, we should only care if the RSS is above the memory target and then shrink the cache if necessary.

Reply:
Ack. I had added it here as a hard bound for when the cache_limit grows above the memory_target. Also for the cases where a user explicitly sets cache_limit during runtime and he sets it above the target unknowingly. Probably should have added a dout message saying that the cache limit has exceeded the target and resetting to to (memory target/1.25). But now that I think about it, the additional checks kind of make it deviate from the target objective: resize according to RSS.